File: RandomForestRegression.py
#Random Forest Regression

#Importing libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

#import dataset
dataset = pd.read_csv("Position_Salaries.csv")
X = dataset.iloc[:, 1:2].values
y = dataset.iloc[:, 2].values

#split data not needed
#feature scaling included in model

#fitting the Random Forest model to the dataset
from sklearn.ensemble import RandomForestRegressor
regressor = RandomForestRegressor(n_estimators = 1000, random_state = 0)
regressor.fit(X,y)

#predicting a new result
y_pred = regressor.predict((6.5))

#visualize RF regression results in high resolution (smooth curve)
X_grid = np.arange(min(X), max(X), 0.01)
X_grid = X_grid.reshape((len(X_grid), 1))
plt.scatter(X,y, color = "red")
plt.plot(X_grid, regressor.predict(X_grid), color = "blue")
plt.title("Truth or Bluff (Random Forest Regression)")
plt.xlabel("Position Level")
plt.ylabel("Salary")
plt.show

File: RandomForestRegression.r
#Random Forest Regression

#importing the dataset
dataset <- read.csv("Position_Salaries.csv")
dataset <- dataset[2:3]

#splitting data into testing and training sets is not necessary
#feature scaling is handled

#fitting the Random Forest Regression model to the dataset
library(randomForest)
set.seed(123)
regressor <- randomForest(x = dataset[1],
                          y = dataset$Salary,
                          ntree = 1000)

#Predicting a new result  
y_pred <- predict(regressor, data.frame(Level = 6.5))  
  
#visualizing the Random Forest Regression Model
library(ggplot2)
x_grid = seq(min(dataset$Level), max(dataset$Level), 0.001)
ggplot()+ 
  geom_point(aes(x = dataset$Level, y = dataset$Salary),
             color = "red") +
  geom_line(aes(x = x_grid, y = predict(regressor, newdata = data.frame(Level = x_grid))),
            color = "blue") +
  ggtitle("Truth or Bluff (Random Forest Regression)") +
  xlab("Level") +
  ylab("Salary")
